From cc794f8c80095eab0db0e2bba71d55b48e7fd2cf Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Wed, 28 Jan 2015 15:45:05 -0800
Subject: [PATCH 0013/1164] CDH-24369 - Do not throw an exception if a hive
 property does not exist

(cherry picked from commit ddb57b5936d2c0f70c54f2ef91a767da8b17f273)
---
 .../hadoop/hive/ql/processors/SetProcessor.java    |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/processors/SetProcessor.java b/ql/src/java/org/apache/hadoop/hive/ql/processors/SetProcessor.java
index bc9254c..3d504e8 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/processors/SetProcessor.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/processors/SetProcessor.java
@@ -29,6 +29,8 @@
 import java.util.SortedMap;
 import java.util.TreeMap;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.apache.hadoop.hive.conf.HiveConf;
 import org.apache.hadoop.hive.metastore.api.FieldSchema;
 import org.apache.hadoop.hive.metastore.api.Schema;
@@ -41,6 +43,7 @@
  *
  */
 public class SetProcessor implements CommandProcessor {
+  private static final Log LOG = LogFactory.getLog(SetProcessor.class);
 
   private static final String prefix = "set: ";
 
@@ -161,7 +164,7 @@ private static void setConf(String varname, String key, String varvalue, boolean
           throw new IllegalArgumentException(message.toString());
         }
       } else if (key.startsWith("hive.")) {
-        throw new IllegalArgumentException("hive configuration " + key + " does not exists.");
+        LOG.warn("hive configuration " + key + " does not exists.");
       }
     }
     conf.verifyAndSet(key, value);
-- 
1.7.9.5

